### PR TITLE
[netcore] Add separate mono-netcore.sln for VS builds

### PR DESCRIPTION
**Pull request by akoeplinger**
Only includes projects that are relevant for netcore.

The build scripts now also look for preview VS versions so you don't need to have both preview and stable VS installed.

### CODE REVIEW

File: msvc/mono-netcore.sln [hunk 1]
```diff
@@ -0,0 +1,220 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (marek-safar) [inline msvc/mono-netcore.sln]: I am not sure this is right location. We don't want to mirror this and I'd prefer netcore sln not to be mirrored at all and live in dotnet/runtime

Comment (akoeplinger) [reply msvc/mono-netcore.sln]: The current plan is to mirror the whole msvc/ directory so it's easiest if this file is here too. The sln file typically doesn't need changes so it shouldn't be an issue.

Comment (marek-safar) [reply msvc/mono-netcore.sln]: I didn't see it in the repo consolidation plan and this actually goes against the prefered policy of mirroring the minimum files possible

Comment (akoeplinger) [reply msvc/mono-netcore.sln]: this is the minimum possible since we don't want to diverge the .vcxproj's (same story as the Makefiles)

**Comment by akoeplinger** [comment]
We're going to keep this in dotnet/runtime only.

**Pull request closed without merging**